internal/service/searchdeployment/resource.go

```diff
@@ -68,25 +67,20 @@ func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resou
 	if diags.HasError() {
 		return
 	}
-	if plan.DeleteOnCreateTimeout.ValueBool() {
-		var deferCall func()
-		deleteOnTimeout := func(newCtx context.Context) error {
-			cleanup.ReplaceContextDeadlineExceededDiags(diags, createTimeout)
-			_, err := connV2.AtlasSearchApi.DeleteAtlasSearchDeployment(newCtx, projectID, clusterName).Execute()
-			return err
-		}
-		ctx, deferCall = cleanup.OnTimeout(
-			ctx, createTimeout, diags.AddWarning, fmt.Sprintf("Search Deployment %s, (%s)", clusterName, projectID), deleteOnTimeout,
-		)
-		defer deferCall()
-	}
 	if _, _, err := connV2.AtlasSearchApi.CreateAtlasSearchDeployment(ctx, projectID, clusterName, &createReq).Execute(); err != nil {
 		diags.AddError("error during search deployment creation", err.Error())
 		return
 	}
 
 	deploymentResp, err := WaitSearchNodeStateTransition(ctx, projectID, clusterName, connV2.AtlasSearchApi,
 		RetryTimeConfig(createTimeout, minTimeoutCreateUpdate))
+	err = cleanup.HandleCreateTimeout(cleanup.ResolveDeleteOnCreateTimeout(plan.DeleteOnCreateTimeout), err, func(ctxCleanup context.Context) error {
+		_, err := connV2.AtlasSearchApi.DeleteAtlasSearchDeployment(ctxCleanup, projectID, clusterName).Execute()
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
```suggestion
		return err
```

you can just return err

Reply:
fixed in [961cf66](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/3595/commits/961cf6633b2ae3c5cbc07d578cc308c66d60a1dc)